File: app/src/main/java/com/example/ftpnext/core/LoadDirection.java
package com.example.ftpnext.core;

public enum LoadDirection {

    DOWNLOAD(0),

    UPLOAD(1),

    DEFAULT(0);

    private final int mValue;

    LoadDirection(int iValue) {
        this.mValue = iValue;
    }

    public int getValue() {
        return mValue;
    }

    public static LoadDirection getValue(int iValue) {
        if (iValue == 0)
            return LoadDirection.DOWNLOAD;
        else if (iValue == 1)
            return LoadDirection.UPLOAD;
        return LoadDirection.DEFAULT;
    }

}

File: app/src/main/java/com/vpulse/ftpnext/core/LoadDirection.kt
package com.vpulse.ftpnext.core

enum class LoadDirection(val value: Int) {
    DOWNLOAD(0),
    UPLOAD(1),
    DEFAULT(0);

    companion object {
        fun getValue(iValue: Int): LoadDirection {
            if (iValue == 0) return DOWNLOAD else if (iValue == 1) return UPLOAD
            return DEFAULT
        }
    }
}